osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/BatchInserter.cs

```diff
@@ -363,85 +306,111 @@ public static async Task<ScoreInfo> CreateReferenceScore(Ruleset ruleset, HighSc
 
 #pragma warning restore CS0618
 
-            Beatmap beatmap = await connection.QuerySingleAsync<Beatmap>("SELECT * FROM osu_beatmaps WHERE `beatmap_id` = @BeatmapId", new
-            {
-                BeatmapId = highScore.beatmap_id
-            }, transaction);
-
-            LegacyBeatmapConversionDifficultyInfo difficulty = LegacyBeatmapConversionDifficultyInfo.FromAPIBeatmap(beatmap.ToAPIBeatmap());
+            var difficulty = getDificultyInfo(highScore.beatmap_id).Result;
```

Comment:
not sure there's much point in `getDifficultyInfo()` being `async` if its only caller is gonna call `.Result`